python/cuml/tsa/arima.pyx

```diff
@@ -294,455 +317,613 @@ class ARIMAModel(Base):
             raise NotImplementedError("IC type '{}' unknown". format(ic_type))
 
         information_criterion(handle_[0], <double*> d_y_ptr,
-                              <int> self.num_batches, <int> self.num_samples,
-                              <int> p, <int> d, <int> q, <double*> d_mu_ptr,
-                              <double*> d_ar_ptr, <double*> d_ma_ptr,
-                              <double*> ic.data(), ic_type_id)
+                              <int> self.batch_size, <int> self.n_obs,
+                              <ARIMAOrder> order, cpp_params,
+                              <double*> ic.data(), <int> ic_type_id)
 
         return ic
 
     @property
     def aic(self):
+        """Akaike Information Criterion"""
         return self._ic("aic")
 
     @property
     def aicc(self):
+        """Corrected Akaike Information Criterion"""
         return self._ic("aicc")
 
     @property
     def bic(self):
+        """Bayesian Information Criterion"""
         return self._ic("bic")
 
-    def predict_in_sample(self):
-        """Return in-sample prediction on batched series given batched model
+    @property
+    def complexity(self):
+        """Model complexity (number of parameters)"""
+        p, _, q = self.order
+        P, _, Q, s = self.seasonal_order
+        return p + P + q + Q + self.intercept + 1
```

Comment:
In the interest of not repeating ourselves, maybe you can just use `self.order.complexity()` call itself?